app/src/main/java/com/team41/boromi/adapters/PagerAdapter.java

```diff
@@ -0,0 +1,43 @@
+package com.team41.boromi.adapters;
+
+import android.os.Bundle;
+import android.util.Pair;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.lifecycle.Lifecycle;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+import com.team41.boromi.FragmentFactory;
+import java.util.ArrayList;
+
+public class PagerAdapter extends FragmentStateAdapter {
+
+  private ArrayList<Pair<Class<? extends Fragment>, Bundle>> fragmentClasses;
+
+  public PagerAdapter(@NonNull FragmentManager fragmentManager,
+      @NonNull Lifecycle lifecycle) {
+    super(fragmentManager, lifecycle);
+    fragmentClasses = new ArrayList<>();
+  }
+
+  @NonNull
+  @Override
+  public Fragment createFragment(int position) {
+    Pair<Class<? extends Fragment>, Bundle> p = fragmentClasses.get(position);
+    try {
+      return FragmentFactory.createFragment(p.first, p.second);
+    } catch (InstantiationException | IllegalAccessException e) {
+      e.printStackTrace();
+      throw new RuntimeException("Failed to create fragment");
```

Comment:
I think this is good, just crash the app and leave a log since we don't know how to handle. This may be a little picky but if you could add a cause by passing e into this runtimeException and maybe leaving a ```log.w("ADAPTERTAG", e)``` I will be satisfied 😄